js/pivx_shield.js

```diff
@@ -197,42 +190,43 @@ export class PIVXShield {
   }
 
   //Save your shield data
-  async save() {
-    const { address, _ } = await this.callWorker(
-      "generate_default_payment_address",
-      this.extfvk,
-      this.isTestNet,
-    );
-
+  save() {
     return JSON.stringify(
       new ShieldData({
-        defaultAddress: address,
+        extfvk: this.extfvk,
         lastProcessedBlock: this.lastProcessedBlock,
         commitmentTree: this.commitmentTree,
         diversifierIndex: this.diversifierIndex,
         unspentNotes: this.unspentNotes,
+        isTestNet: this.isTestNet,
       }),
     );
   }
 
   /**
-   * Load shieldWorker from a shieldData
+   * Creates a PIVXShield object from a ShieldData
    * @param {ShieldData} shieldData - shield data
    */
```

Comment:
```suggestion
   * Creates a PIVXShield object from a ShieldData
   * @param {string} shieldData - shield data
   */
```
Change this to string and add `JSON.parse` inside the function. We are not making shieldData transparent to the user, so I would avoid passing the object directly.

Reply:
done